internal/debug/debug.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug // import "go.opentelemetry.io/otel/internal/debug"
+
+import (
+	"log"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/stdr"
+)
+
+// globalLogger is the logging interface used within the otel api and sdk provide deatails of the internals.
+//
+// The default logger uses stdr which is backed by the standard `log.Logger`
+// interface. This logger will only show messages at the Error Level.
+var globalLoggger logr.Logger = stdr.New(log.New(os.Stderr, "", log.LstdFlags|log.Lshortfile))
```

Comment:
Can we have a null (no-op) logger here?
As a rule of thumb, we should not affect the instrumented application in a "visible" way.

Reply:
This is done so we can enable error reporting, via ErrorHandeler. We should change that default first.

Reply:
> This is done so we can enable error reporting, via ErrorHandeler. 

Cannot the user enable it by himself via `otel.SetErrorHandler` ?

> We should change that default first.

I do not follow 😞 Can you rephrase

I think that it would be better just to include in this PR an internal API for logging. And not touch the ErrorHandler at all.

Reply:
Resolving per https://github.com/open-telemetry/opentelemetry-go/issues/2417